src/atoms/forms/canvas-sketch/react-canvas-sketch.tsx

```diff
@@ -0,0 +1,219 @@
+                backgroundImageCanvas: htmlCanvasBackgroundImage.current as HTMLCanvasElement,
+                currentObjectIndex: props.value.currentObjectIndex,
+                objectStack: props.value.objects,
+                panX: 0,
+                panY: 0,
+                scaleFactor: 1,
+                sketchCanvas: htmlCanvasSketch.current as HTMLCanvasElement,
+            };
+
+            // initialize the canvas sketch object
+            const newCanvasSketch = new CanvasSketch(canvasSketchConfig);
+
+            // set up any default property driven settings
+            newCanvasSketch.setSelectedTool(props.canvasToolType);
+            newCanvasSketch.setToolColor(props.toolColor);
+            newCanvasSketch.setToolWidth(props.toolWidth);
+            newCanvasSketch.setOnAddedToolStroke(props.onAddedStroke);
+
+            // set state
+            setCanvasSketch(newCanvasSketch);
+            setIsInitialized(true);
+        }
+    }, [
+        canvasSketch,
+        htmlCanvasBackgroundImage,
+        htmlCanvasSketch,
+        isInitialized,
+        props.backgroundImageUrl,
+        props.canvasToolType,
+        props.onAddedStroke,
+        props.toolColor,
+        props.toolWidth,
+        props.value.currentObjectIndex,
+        props.value.objects,
+    ]);
+
+    useEffect(() => {
+        console.log("useEffect: canvasSketch");
+        return () => {
+            // cleanup on unmount
+            if (canvasSketch != null) {
+                canvasSketch.dispose();
+            }
+        }
+    }, [canvasSketch]);
+
+    useEffect(() => {
+        if (canvasSketch == null) {
+            return;
+        }
+        canvasSketch.redrawCurrentState();
+
+    }, [
+        canvasSketch,
+        props.canvasHeight,
+        props.canvasWidth,
+        props.containerHeight,
+        props.containerWidth,
+    ]);
+
+    useEffect(() => {
+        if (canvasSketch == null) {
+            return;
+        }
+        // Only redraw if currentObjectIndex changes but not the objects list itself.  Being done to
+        // prevent rerenders after every single stroke of the same tool.  See following URL for more
+        // inforamtion on tracking previous state/props using hooks
+        //      https://stackoverflow.com/questions/55724642/react-useeffect-hook-when-only-one-of-the-effects-deps-changes-but-not-the-oth
+        if (prevValueObjects === props.value.objects) {
+            canvasSketch.redrawSketchAt(props.value.objects, props.value.currentObjectIndex);
+        }
+    }, [props.value.currentObjectIndex, props.value.objects, canvasSketch, prevValueObjects]);
+
+    useEffect(() => {
+        if (canvasSketch == null) {
+            return;
+        }
+        canvasSketch.redrawSketchAt(props.value.objects, props.value.currentObjectIndex);
+    }, [props.redrawIncrement, canvasSketch, props.value.objects, props.value.currentObjectIndex]);
```

Comment:
This effect hook is triggered anytime the renderer of `<ReactCanvasSketch/>` passes in a different `redrawIncrement` integer where as the other effect hook doesn't have a dependency on `redrawIncrement`.  This is a way I've allowed high level components control their rendered subordinates by way of props instead of tracking refs which I avoid as much as possible.